### PR TITLE
[SPARK-11871] Add save/load for MLPC

### PR DESCRIPTION
**Pull request by yinxusen**
## What changes were proposed in this pull request?

https://issues.apache.org/jira/browse/SPARK-11871

Add save/load for MLPC
## How was this patch tested?

Test with Scala unit test

### CODE REVIEW

**Comment by yinxusen** [comment]
Pls ignore my first comment, I have splited the test of model and classifier separately.

**Comment by mengxr** [comment]
cc @avulanov

**Comment by avulanov** [comment]
LGTM

**Comment by yinxusen** [comment]
@mengxr @jkbradley I change the version code. Does it looks good to you?

**Comment by jkbradley** [comment]
I'll take a look now!

**Comment by jkbradley** [comment]
In the meantime, could you please update the description (to be a reasonable commit message)?  Thanks

File: mllib/src/main/scala/org/apache/spark/ml/classification/MultilayerPerceptronClassifier.scala [hunk 1]
```diff
@@ -214,4 +224,56 @@ class MultilayerPerceptronClassificationModel private[ml] (
...
+    override protected def saveImpl(path: String): Unit = {
+      // Save metadata and Params
+      DefaultParamsWriter.saveMetadata(instance, path, sc)
+      // Save model data: layers, weights
+      val data = Data(instance.layers, instance.weights)
+      val dataPath = new Path(path, "data").toString
+      sqlContext.createDataFrame(Seq(data)).repartition(1).write.parquet(dataPath)
+    }
+  }
+  private class MultilayerPerceptronClassificationModelReader
```
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/ml/classification/MultilayerPerceptronClassifier.scala]: newline

File: mllib/src/test/scala/org/apache/spark/ml/classification/MultilayerPerceptronClassifierSuite.scala [hunk 2]
```diff
@@ -91,4 +100,26 @@ class MultilayerPerceptronClassifierSuite extends SparkFunSuite with MLlibTestSp
...
+      .setLayers(Array(2, 3, 2))
+      .setMaxIter(5)
+      .setBlockSize(2)
+      .setSeed(42)
+      .setTol(0.1)
+      .setFeaturesCol("myFeatures")
+      .setLabelCol("myLabel")
+      .setPredictionCol("myPrediction")
+
+    testDefaultReadWrite(mlp)
```
Comment (jkbradley) [inline mllib/src/test/scala/org/apache/spark/ml/classification/MultilayerPerceptronClassifierSuite.scala]: Can you set argument ```testParams = true``` here?

File: mllib/src/test/scala/org/apache/spark/ml/classification/MultilayerPerceptronClassifierSuite.scala [hunk 3]
```diff
@@ -91,4 +100,26 @@ class MultilayerPerceptronClassifierSuite extends SparkFunSuite with MLlibTestSp
...
+  }
+
+  test("read/write: MultilayerPerceptronClassificationModel") {
+    val mlp = new MultilayerPerceptronClassifier().setLayers(Array(2, 3, 2)).setMaxIter(5)
+    val mlpModel = mlp.fit(dataset)
+    val newMlpModel = testDefaultReadWrite(mlpModel)
```
Comment (jkbradley) [inline mllib/src/test/scala/org/apache/spark/ml/classification/MultilayerPerceptronClassifierSuite.scala]: Can you set argument ```testParams = true``` here?

**Comment by jkbradley** [comment]
The style issue & those little comments are the only items I see.  Thanks!

**Comment by yinxusen** [comment]
test it please

**Comment by yinxusen** [comment]
test it please

**Comment by yinxusen** [comment]
@jkbradley Fixed.

**Comment by yinxusen** [comment]
@jkbradley Is this looks good now?

**Comment by mengxr** [comment]
test this please

**Comment by mengxr** [comment]
LGTM. Merged into master. Thanks!

**Comment by yinxusen** [comment]
Thanks!

**Comment by yanboliang** [comment]
Send #11952 for Python API.

**Pull request closed without merging**